# motor name. should match an actuator name in a transmission interface
test_joint_motor:
  # epos's node information (must be enough to identify the node)
  device: 'EPOS4' # default: 'EPOS4'
  protocol_stack: 'MAXON SERIAL V2' # default: 'MAXON SERIAL V2'
  interface: 'USB' # default: 'USB'
  port: 'USB0' # default: '' (any port)
  node_id: 1 # default: 0 (any node id)
  serial_number: '0x602137010479' # epos's serial number in hex (default: '0' (any number))

  # communication settings (optional)
  # ignored if another node belonging to the same device is already initialized.
  # 'device' is a set of types of device, protocol_stack, interface, and port.
  # order of node initialization is that of commandline arguments.
  baudrate: 1000000 # default: 0 (keep current baudrate)
  timeout: 500 # [ms], default: 0 (keep current timeout)

  # general parameters (optional)
  clear_faults: true # clear faults recorded in epos on startup (default: false)
  rw_ros_units: true # use ros standard units (rad, rad/s, Nm) in hardware interfaces
                     # or epos standard units (quad count of encoder pulse(qc), rpm, mNm)
                     # (default: false)
  detailed_diagnostic: false # additionally read actual operation mode, device status,
                             # and fault info (default: false)

  # map from ros_control's controller to epos's operation mode (required)
  operation_mode_map: 
    'velocity_controller': 'profile_velocity'
    'position_controller': 'profile_position'
    'effort_controller': 'cyclic_synchronoust_torque'

  # motor parameters (required)
  motor:
    type: 11 # 1: DC, 10: EC sinus, 11: EC block
    nominal_current: 2.48 # [A]
    max_output_current: 33.7 # [A]
    torque_constant: 22.8 # [mNm/A]
    thermal_time_constant: 31.2 # thermal time constant of motor widing [s]
    number_of_pole_pairs: 2 # EC only

  # sensor parameters (required)
  sensor:
    type: 1 # 1: INC 3CH, 2: INC 2CH, 4: SSI ABS BIN, 5: SSI ABS GRY
    resolution: 1024 # encoder resolution
    inverted_polarity: false

  # safety parameters (required)
  safety:
    max_following_error: 20000 # permitted error [qc] in profile_position mode
    max_profile_velocity: 8800 # max velocity [rpm] in profile_position and profile_velocity modes
    max_acceleration: 15000 # max acceleration [rpm/s] in profile_position nad profile_velocity modes

  # profile position mode configs (optional)
  position_profile:
    velocity: 8000 # [rpm]
    acceleration: 8000 # [rpm/s]
    deceleration: 8000 # [rpm/s]

  # profile velocity mode configs (optional)
  velocity_profile:
    acceleration: 8000 # [rpm/s]
    deceleration: 8000 # [rpm/s]

  # power supply parameters (optional)
  power_supply:
    name: 'battery' # if this parameter exists, a battery hardware interface will be registered
    location: 'battery_slot' # corresponds BatteryState::location
    technology: 2 # corresponds BatteryState::power_supply_technology
    serial_number: 'unknown' # corresponds BatteryState::serial_number
